core/src/scene/styleParam.cpp

```diff
@@ -294,11 +282,14 @@ StyleParam::Value StyleParam::parseString(StyleParamKey key, const std::string&
     case StyleParamKey::outline_visible:
     case StyleParamKey::collide:
     case StyleParamKey::text_optional:
-    case StyleParamKey::text_collide:
-        if (_value == "true") { return true; }
-        if (_value == "false") { return false; }
-        LOGW("Invalid boolean value %s for key %s", _value.c_str(), StyleParam::keyName(key).c_str());
+    case StyleParamKey::text_collide: {
+        bool result = false;
+        if (YAML::convert<bool>::decode(node, result)) {
```

Comment:
I vaguely remember discussion around, `true`/`false` and `yes`/`no` and other boolean yaml values.

Can you confirm if `YAML::convert<bool>::decode(node, result)` will only handle `true`/`false` values?

Reply:
It accepts only `true`, `True`, `TRUE`, `false`, `False`, and `FALSE`. Will update to use the `YamlUtil` version of this, for consistency.